src/types/domain-names.ts

```diff
@@ -0,0 +1,8 @@
+import type { Stage } from "../constants";
+
+export type GuDomainNameProps = Record<Stage, GuDomainName>;
+
+export interface GuDomainName {
+  domainName: string;
+  hostedZoneId?: string;
```

Comment:
The `hostedZoneId?` is here because:

1. `GuCertificate` and `GuCname` share this interface 
2. We need the hosted zone id for the former if the DNS record required for certificate validation will reside in `Route53`.

It's slightly tangential to this PR, but it's a bit odd that we offer proper support for creating the certificate in a Route 53 zone, but we _don't_ offer the same support for managing your CNAME... Perhaps we should consider making the `GuCname` construct DNS-provider agnostic in the future?

Reply:
> It's slightly tangential to this PR, but it's a bit odd that we offer proper support for creating the certificate in a Route 53 zone, but we _don't_ offer the same support for managing your CNAME... Perhaps we should consider making the `GuCname` construct DNS-provider agnostic in the future?

Should we drop Route53 support entirely as Route53 isn't on our paved road? AWS's constructs for Route53 are pretty good too, so being agnostic might yield more complicated code?

I've not thought about this that much though - not blocking this PR but would love to discuss.

Reply:
> AWS's constructs for Route53 are pretty good too, so being agnostic might yield more complicated code?

Yes, you're probably right. I know that we tried to do this before, but I guess we need to understand more generally whether we should use/support Route53 in the department. It would certainly be simpler to have a single DNS provider, but I worry that we have so much stuff in Route53 now that migrating away might be unfeasible.

Reply:
> not blocking this PR but would love to discuss.

Definitely agree that we should discuss this more outside of the PR though!